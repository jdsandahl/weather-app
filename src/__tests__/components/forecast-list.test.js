import React from 'react';
import { cleanup, render } from '@testing-library/react';
import ForecastList from '../../components/forecast-list';
import moment from 'moment';

afterEach(cleanup);

describe("ForecastList", () => {
    const forecasts = [
        {
            date: 123456789,
            description: "date1",
            icon: "200",
            temperature: {
                max: 999,
            },
        },
        {
            date: 456,
            description: "date2",
            icon: "200",
            temperature: {
                max: 888,
            },
        },
    ];

    it("renders correctly to match ForecastList snapshot", () => {
        const { asFragment } = render(<ForecastList forecasts={ forecasts } />);
    
        expect(asFragment()).toMatchSnapshot();
    });

    it("renders the correct amount of ForecastSummary component props", () => {
        const { getAllByTestId } = render (<ForecastList forecasts={ forecasts } />);

        expect(getAllByTestId("date-id")).toHaveLength(2);
        expect(getAllByTestId("icon-id")).toHaveLength(2);
        expect(getAllByTestId("temperature-id")).toHaveLength(2);
        expect(getAllByTestId("description-id")).toHaveLength(2);
    });

    it("renders the correct dates and in the correct format", () => {
        const { getAllByText } = render (<ForecastList forecasts={ forecasts } />);
        
        const testDate1 = moment(123456789).format('ddd Do MMM');
        const testDate2 = moment(456).format('ddd Do MMM');

        expect(getAllByText(testDate1)).toBeTruthy();
        expect(getAllByText(testDate1)).toHaveLength(1);

        expect(getAllByText(testDate2)).toBeTruthy();
        expect(getAllByText(testDate2)).toHaveLength(1);
    });
});