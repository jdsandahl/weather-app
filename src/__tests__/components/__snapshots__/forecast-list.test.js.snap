// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ForecastList renders correctly to match ForecastList snapshot 1`] = `
<DocumentFragment>
  <div
    class="forecast-list"
  >
    <div
      class="forecast-summary"
    >
      <div
        class="date"
        data-testid="date-id"
      >
        <span>
          Fri 2nd Jan
        </span>
      </div>
      <div
        class="icon"
        data-testid="icon-id"
      >
        <i
          class="wi wi-day-storm-showers"
          name="owm"
        />
      </div>
      <div
        class="temperature"
        data-testid="temperature-id"
      >
        <span>
          999°c
        </span>
      </div>
      <div
        class="description"
        data-testid="description-id"
      >
        <span>
          date1
        </span>
      </div>
      <button>
        More Details
      </button>
    </div>
    <div
      class="forecast-summary"
    >
      <div
        class="date"
        data-testid="date-id"
      >
        <span>
          Thu 1st Jan
        </span>
      </div>
      <div
        class="icon"
        data-testid="icon-id"
      >
        <i
          class="wi wi-day-storm-showers"
          name="owm"
        />
      </div>
      <div
        class="temperature"
        data-testid="temperature-id"
      >
        <span>
          888°c
        </span>
      </div>
      <div
        class="description"
        data-testid="description-id"
      >
        <span>
          date2
        </span>
      </div>
      <button>
        More Details
      </button>
    </div>
  </div>
</DocumentFragment>
`;
